Reformatted code to follow Eslint

justinhungle opened the pull request:
I also fixed a reference error in shared/api.js for getProducts.

saarikak commented on client/src/shared/api.js:
```diff
@@ -1,35 +1,18 @@
+  return axios.get('/api/catwalk/products');
 };
 
-export const getReviews = (productId) => {
-  return axios.get('/api/catwalk/reviews', {
-    params: { id: productId }
```
Reviews and potentially Q&A may need to send other params. Should we allow for a params object to be sent as an argument instead of productId and refactor the code to handle that?

justinhungle replied:
I think that's a good idea. I wasn't thinking of that

saarikak commented on server/controllers/controllers.js:
```diff
@@ -1,63 +1,63 @@
   getProducts: (req, res) => {
     const params = req.query.id;
     models.getProducts(params, (error, results) => {
-      if(error) {
+      if (error) {
         res.status(500).send(error);
       } else {
-        res.send(results)
-      };
+        res.send(results);
+      }
     });
```
Potentially refactor all functions with promises if returning axios.get from models.js as suggested

justinhungle replied:
Agreed

saarikak reviewed (changes requested):
Overall, the shared server code looks great! I added in some comments to allow code to follow DRY principles as well as to proactively deal with additional parameters required for API calls.

justinhungle commented on server/models/models.js:
```diff
@@ -1,97 +1,97 @@
-      })
-      .catch((error) => {
-        callback(error);
+        headers: {
+          Authorization: `${config.access_token}`,
+        },
```
Sounds good will refactor tomorrow

justinhungle commented on server/models/models.js:
```diff
@@ -1,97 +1,97 @@
-    .then((results) => {
-      callback(null, results.data);
-    })
-    .catch((error) => {
-      callback(error);
+        Authorization: `${config.access_token}`,
+      },
     })
+      .then((results) => {
+        callback(null, results.data);
+      })
+      .catch((error) => {
+        callback(error);
+      });
```
That's a great idea!

justinhungle commented:
@saarikak Hi Saarika,

I've made the changes as requested. 
 - Refactored model functions to return axios promise
 - Refactored controller functions to handle the returned promises from the corresponding model functions
 - Changed input param to an object for api functions getReview, getAnswers, getQuestions and getProducts
Let me know if I missed anything.

saarikak commented on server/controllers/controllers.js:
```diff
@@ -1,63 +1,63 @@
+      });
   },
   getReviews: (req, res) => {
     const params = req.query.id;
```
Only the ID is being utilized and sent to models.js but we may have other params sent from the front-end, so we may need to refactor to send all of the params to the API.

saarikak reviewed (changes requested):
All the changes looked great! I noticed that although we are accepting a params object in api.js, we are not sending anything but the ID param to the API. We may need to refactor to send all params.

saarikak reviewed (approved):
All changes look great!
